migrations/3_deploy_genericHandlerV1.js

```diff
@@ -3,41 +3,39 @@
  * SPDX-License-Identifier: LGPL-3.0-only
  */
 
-const fs = require("fs");
-
 const Helpers = require('../test/helpers');
-
-const networksConfig = JSON.parse(fs.readFileSync("./networks_config.json"));
+const Utils = require('./utils');
 
 const BridgeContract = artifacts.require("Bridge");
-const TestStoreContract = artifacts.require("TestStore");
 const GenericHandlerV1Contract = artifacts.require("GenericHandlerV1");
 const FeeRouterContract = artifacts.require("FeeHandlerRouter");
 const BasicFeeHandlerContract = artifacts.require("BasicFeeHandler");
+const FeeHandlerWithOracleContract = artifacts.require("FeeHandlerWithOracle");
 
 module.exports = async function(deployer, network) {
+    const networksConfig = Utils.getNetworksConfig()
     // trim suffix from network name and fetch current network config
     let currentNetworkName = network.split("-")[0];
     let currentNetworkConfig = networksConfig[network.split("-")[0]];
+    delete networksConfig[currentNetworkName]
 
     // fetch deployed contracts addresses
     const bridgeInstance = await BridgeContract.deployed();
     const feeRouterInstance = await FeeRouterContract.deployed();
     const basicFeeHandlerInstance = await BasicFeeHandlerContract.deployed();
+    const feeHandlerWithOracleInstance = await FeeHandlerWithOracleContract.deployed();
 
     // deploy generic handler
     const genericHandlerV1Instance = await deployer.deploy(GenericHandlerV1Contract, bridgeInstance.address);
```

Comment:
Lets get rid of this V1 staff, I am really confused all the time to understand what generic we are talking about.
As I remember we agreed that now we should have only 2 generics. Permission Generic Handler (old one) and Permissionless Generic handler (that is not yet done so we can skip it in this migration)

Reply:
This is basically the permissionless. I would not get rid of it. Migrations should always track the current state of the repo and be updated with it. Removing and then returning it would just make it more confusing.

Renaming it to permissionless (which I think Nikola will do in his PR) will have more sense and remove the confusion.

Reply:
This is generally the reason why I wanted to separate the migrations. Because we do not have anything on a TestNet...

Reply:
Don't really understand what do you mean.

Reply:
`GenericHandlerV1Contract` will be renamed in a PR that will address [Handlers interface](https://app.zenhub.com/workspaces/sygma-protocol-62f126a54b7d5a55203732f0/issues/sygmaprotocol/sygma-solidity/70), then it will be called `PermissionedGenericHandler and `PremisionlessGenericHandlerì`.
That make most sense to as it would require multiple updates to naming/scripts/contracts if done earlier.

Reply:
I mean that now we do not want to remove old Generic handler (hackathon one) from migrations since it is used in Develop, but we do not need it  for the TestNet

Reply:
Don't see it as a problem TBH

Reply:
but we will deploy unnecessary contracts and set it up on testnet, even though it will never be used.
The same will be for the main net i assume

Reply:
I thought about adding a check if a contract is setup in config to deploy it or not, but decided against it adding only a check that it isn't registered if there is no resource ID set because usually when we are deploying we want the latest of everything and the few contracts that we don't want will just be deployed and unused which doesn't impact anything except a minor increase in gas.

Reply:
But when we will remove this V1 contract from repo?

Reply:
When we make the new version.
It is called V1, but we are planning to keep multiple versions.
There will be only one permissionless handler